// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package account;

// The Account service definition.
service Account {
  rpc GetAccountById (AccountRequestById) returns (AccountResponse) {}
  rpc GetAccountByUsername (AccountRequestByUsername) returns (AccountResponse) {}
  rpc UpdateAccount (UpdateAccountRequest) returns (AccountResponse) {}
  rpc CreateAccount (CreateAccountRequest) returns (AccountResponse) {}
  rpc DeleteAccount (UpdateAccountRequest) returns (DeleteResponse) {}
}
message AccountRequestById {
  string userId = 1;
}
message AccountRequestByUsername{
  string username = 1;
}
message CreateAccountRequest{
  string firstname = 1;
  string lastname = 2;
  string username = 3;
  string password = 4;
  string email = 5;
  int32 country = 6;
  string address = 7;
  int32 user_type_id = 8;
}
message UpdateAccountRequest {
  string userId = 1;
  string firstname = 2;
  string lastname = 3;
  string username = 4;
  string password = 5;
  string email = 6;
  int32 country_id = 7;
  string address = 8;
}
message AccountResponse {
  string userId = 1;
  string firstname = 2;
  string lastname = 3;
  string username = 4;
  string password = 5;
  string email = 6;
  int32 country_id = 7;
  string address = 8;
  int32 user_type_id = 9;
}
message DeleteResponse{
  int32 status = 1;
  string message = 2;
}